lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3893,6 +3894,20 @@ namespace {
       recordObjCOverride(result);
     }
 
+    Decl *VisitCXXDestructorDecl(const clang::CXXDestructorDecl *decl) {
+      // If the destructor doesn't exist or is provided by the user, we won't
+      // need to define it.
+      if (decl->isUserProvided() || decl->isDeleted())
+        return nullptr;
+      // Otherwise, we're going to need to tell sema to define the destructor.
+      auto &clangSema = Impl.getClangSema();
+      clangSema.DefineImplicitDestructor(clang::SourceLocation(),
```

Comment:
I think you may also need `DeclareImplicitDestructor`? That requires a `CXXRecordDecl`, so it's probably easiest to put all of the code in `VisitCXXRecordDecl` (the same way it's done for constructors).

Reply:
Now that you say that, I would have thought the same thing. But, it seems to work without a call to `DeclareImplicitDestructor`. Should I add one anyway? WDYT? 

I'll move this to `VisitCXXRecordDecl` either way.